src/hist.jl

```diff
@@ -27,6 +27,14 @@ end
 end
 
 
+# Need a generated function to promote edge types, because a simple
+# promote_type(map(eltype, h.edges)...) isn't type stable (tested
+# with Julia v0.5).
+@generated function _promote_edge_types{N}(edges::NTuple{N,AbstractVector})
```

Comment:
I don't really understand why it's not (or why the `map` wasn't) - but I think we can live with the generated _promote_edge_types (esp. since it got rid of another generated function).